Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/intersection.h

```diff
@@ -26,9 +26,769 @@
+ * geometric traits that templates`TriangleMesh`.
+ * \tparam OutputIterator a model of `OutputIterator` holding objects of type
+ *   `std::pair<std::size_t, std::size_t>`. This OutputIterator will hold the position of the
+ *  elements in their respective range.
+ * \tparam NamedParameters a sequence of \ref namedparameters
+ *
+ * \param mesh the triangulated surface mesh to be checked.
+ * \param polyline the `Polyline` to check for intersections.
+ * \param out output iterator to be filled with all pairs of face-segment that intersect
+ * \param np optional sequence of \ref namedparameters for `mesh`, among the ones listed below
+ *
+ * \cgalNamedParamsBegin
+ *    \cgalParamBegin{vertex_point_map} the property map with the points associated to the vertices of `pmesh`.
+ *   If this parameter is omitted, an internal property map for
+ *   `CGAL::vertex_point_t` should be available in `TriangleMesh`\cgalParamEnd
+ *    \cgalParamBegin{geom_traits} an instance of a geometric traits class, model of `PMPSelfIntersectionTraits` \cgalParamEnd
+ * \cgalNamedParamsEnd
+ * \return `out`
+ */
+template <class TriangleMesh
+          , class Polyline
+          , class OutputIterator
+          , class NamedParameters
+          >
+OutputIterator
+compute_face_polyline_intersection(const TriangleMesh& mesh,
+              const Polyline& polyline,
+              OutputIterator out,
+              const NamedParameters& np)
+{
+  return compute_face_polyline_intersection(faces(mesh), polyline, mesh, out, np);
+}
+}// namespace internal
+
 namespace Polygon_mesh_processing{
+/**
+ * \ingroup PMP_intersection_grp
+ * tests if two `Polyline`s intersect.
+ * This function depends on the package \ref PkgBoxIntersectionDSummary
+ *
+ * \tparam Polyline a `RandomAccessRange` of `CGAL::Point_3<Kernel>`.
+ * \tparam Kernel a model of `Kernel`.
+ *
+ * @param polyline1 the first `Polyline` to be tested.
+ * @param polyline2 the second `Polyline` to be tested.
+ * @param K an instance of `Kernel`.
+ *
+ * @return true if `polyline1` and `polyline2` intersect
+ */
+template <class Polyline>
+bool do_intersect( const Polyline& polyline1,
+                   const Polyline& polyline2,
+#ifndef DOXYGEN_RUNNING
+                   const typename boost::enable_if<
+                   typename boost::has_range_iterator<Polyline>::type
+                   >::type* = 0
+#endif
+    )
+{
+  typedef typename boost::range_value<Polyline>::type Point;
+  typedef typename CGAL::Kernel_traits<Point>::Kernel K;
+  try
+  {
+    typedef boost::function_output_iterator<CGAL::internal::Throw_at_first_output> OutputIterator;
+    internal::compute_polyline_polyline_intersection(polyline1, polyline2, OutputIterator(), K());
+  }
+  catch( CGAL::internal::Throw_at_first_output::Throw_at_first_output_exception& )
+  { return true; }
+
+  return false;
+}
+
+/**
+ * \ingroup PMP_intersection_grp
+ * tests if two triangulated surface meshes intersect.
```

Comment:
the usual formation is: returns `true` if any pair of faces from `mesh1` and `mesh2` intersect, and `false` otherwise. This at the same time settle the question of inclusion of meshes.